File: app/src/main/java/com/falcon/restaurants/room/meal/MealDao.java
package com.falcon.restaurants.room.meal;

import androidx.room.Dao;
import androidx.room.Query;
import com.falcon.restaurants.room.BaseDao;
import java.util.List;
import io.reactivex.Observable;
import io.reactivex.Single;

@Dao
public abstract class MealDao extends BaseDao<Meal> {

    private static final String TAG = "MealDao";

    @Query("SELECT * from Meal WHERE restaurantId=:typeIdV ORDER BY updatedAt ASC")
    public abstract Observable<List<Meal>> getByRestaurantId(String typeIdV);

    @Query("SELECT * from Meal WHERE id=:mealIdV")
    public abstract Single<Meal> getMealById(String mealIdV);

    //*** getMaxUpdated
    @Query("SELECT MAX(updatedAt) from Meal WHERE active=1")
    public abstract String getMaxUpdatedAt();

    //**** check exists
    @Query("SELECT EXISTS(SELECT * FROM Meal WHERE id = :idV)")
    public abstract Single<Boolean> checkExists(String idV);

    @Override
    public boolean checkExists(Meal meal){
        return checkExists(meal.getId()).blockingGet();
    }

}

File: app/src/main/java/com/falcon/restaurants/room/meal/MealDao.kt
package com.falcon.restaurants.room.meal

import androidx.room.Dao
import androidx.room.Query
import com.falcon.restaurants.room.BaseDao
import io.reactivex.Observable
import io.reactivex.Single

@Dao
abstract class MealDao : BaseDao<Meal>() {
    
    @Query("SELECT * from Meal WHERE restaurantId=:typeIdV ORDER BY updatedAt ASC")
    abstract fun getByRestaurantId(typeIdV: String): Observable<List<Meal>>

    @Query("SELECT * from Meal WHERE id=:mealIdV")
    abstract fun getMealById(mealIdV: String): Single<Meal> 

    //*** getMaxUpdated
    @Query("SELECT MAX(updatedAt) from Meal WHERE active=1")
    abstract override fun getMaxUpdatedAt(): String

    //**** check exists
    @Query("SELECT EXISTS(SELECT * FROM Meal WHERE id = :idV)")
    abstract fun checkExists(idV: String): Single<Boolean>

   override fun checkExists(meal: Meal): Boolean{
        return checkExists(meal.id).blockingGet()
    }

}
